### ISSUE TITLE
[QUESTION] how to change default file icons?

### ISSUE THREAD

**Issue by babygau** [domtronn/all-the-icons.el#31]
hi,

i'm wondering how to change default icons, for example, i want to change `org` file icon to `readme` file icon.

many thanks in advance

**Comment by domtronn** [comment]
Hi there, there are a couple of way but it depends on how this package is being used... I'm assuming this problem is when I'm _guessing_ what icon you want :) This happens in one of two functions;

• `all-the-icons-icon-for-file`
• `all-the-icons-icon-for-mode`

The first one tries to guess the icon for a _filename_ which matches the name against regexps _(this is used by NeoTree)_, whilst the second tries to guess it based on its _major mode_, _.e.g_ `org-mode` _(this is used for spaceline/mode line stuff)_.

Each of these have an `alist` that they match against, for example see [here](https://github.com/domtronn/all-the-icons.el/blob/master/all-the-icons.el#L136)

```el
 ;; This is in `all-the-icons-icon-alist`
 ("^readme"  all-the-icons-octicon "book"  :height 1.0 :v-adjust 0.0 :face all-the-icons-lcyan)
 ("\\.org$"  all-the-icons-fileicon "org"  :face all-the-icons-lgreen)
```

This says that any file that starts with _"readme"_ will use the book icon and any file that has the _".org"_ extension will use the org icon.

So, if you want to always use the book icon for all org files, you would want to override this in `all-the-icons-icon-icon-alist` by adding an overriding matcher to `all-the-icons-icon-alist` like this;

_**tl;dr**_ Add this to you config
```el
(add-to-list 'all-the-icons-icon-alist '("\\.org$" all-the-icons-octicon "book"))
```

The element you're adding should go `'(REGEX   INSERT_FUNCTION   ICON_NAME)` and you can call `all-the-icons-insert` to find the icon name and insert function you want to use :)

Sorry this is a lot of information! But I hope it helps, and if you have any more questions let me know.

**Comment by babygau** [comment]
it's awesome @domtronn , thank you for your thorough explanation, it worked!
